http/src/main/resources/org/broadinstitute/dsde/workbench/leonardo/liquibase/changesets/20200123_cluster_table_gce_refactor.xml

```diff
@@ -0,0 +1,84 @@
+
+    <changeSet logicalFilePath="leonardo" author="qi" id="cluster_table_gce_refactor">
+        <preConditions onFail="MARK_RAN">
+            <not>
+                <tableExists tableName="RUNTIME_CONFIG"/>
+            </not>
+        </preConditions>
+
+        <dropNotNullConstraint columnName="masterMachineType" tableName="CLUSTER" columnDataType="varchar(256)"/>
+        <dropNotNullConstraint columnName="masterDiskSize" tableName="CLUSTER" columnDataType="INT"/>
+        <dropNotNullConstraint columnName="numberOfWorkers" tableName="CLUSTER" columnDataType="INT"/>
+
+<!--                  Create RUNTIME_CONFIG table   -->
+        <createTable tableName="RUNTIME_CONFIG">
+            <column name="id" type="BIGINT" autoIncrement="true">
+                <constraints primaryKey="true"/>
+            </column>
+            <column name="clusterId" type="BIGINT"></column>
+            <column name="cloudService" type="ENUM('DATAPROC', 'GCE')" defaultValue="DATAPROC">
+                <constraints nullable="false"/>
+            </column>
+
+            <column name="machineType" type="varchar(256)" defaultValue="n1-standard-4">
+                <constraints nullable="false"/>
+            </column>
+
+            <!-- This might change when we move to PD -->
+            <column name="diskSize" type="INT" defaultValue="500">
+                <constraints nullable="false"/>
+            </column>
+
+            <!-- dataproc only -->
+            <column name="numberOfWorkers" type="INT" defaultValue="0">
+                <constraints nullable="false"/>
+            </column>
+
+            <!-- dataproc only -->
+            <column name="workerMachineType" type="varchar(256)" defaultValue="NULL" />
+
+            <!-- dataproc only -->
+            <column name="workerDiskSize" type="INT" defaultValue="NULL" />
+
+            <!-- dataproc only -->
+            <column name="numberOfWorkerLocalSSDs" type="INT" defaultValue="NULL" />
+
+            <!-- dataproc only -->
+            <column name="numberOfPreemptibleWorkers" type="INT" defaultValue="NULL"/>
+
+            <column name="dateAccessed" type="TIMESTAMP(6)"/>
+        </createTable>
+
+        <sql>insert into RUNTIME_CONFIG (`clusterId`, `machineType`, `diskSize`, `numberOfWorkers`, `workerMachineType`, `workerDiskSize`, `numberOfWorkerLocalSSDs`, `numberOfPreemptibleWorkers`) select id, masterMachineType, masterDiskSize, numberOfWorkers, workerMachineType, workerDiskSize, numberOfWorkerLocalSSDs, numberOfPreemptibleWorkers from CLUSTER</sql>
```

Comment:
I am curious how long this takes on dev data

Reply:
Also is there a way to run this in a transaction so we're not left with partial results if it fails? Not sure what liquibase supports.

Reply:
I added dropping columns as a separate changeSet which has precondition that every cluster id should have a record in RUNTIME_CONFIG, if it fails, things will HALT...We also backup database every few hours in production.

I tried a new dev database clone, and it has 20k records, whereas production only has 15406 in CLUSTER table, so I think we should be fine.